devimint/src/util.rs

```diff
@@ -419,3 +420,153 @@ impl GatewayLndCli {
         get_command_for_alias("FM_GWCLI_LND", "gateway-lnd")
     }
 }
+
+pub struct FedimintdCmd;
+impl FedimintdCmd {
+    pub async fn cmd(self) -> Command {
+        get_command_for_alias("FM_FEDIMINTD_BASE_EXECUTABLE", "fedimintd")
+    }
+}
+
+/// Representation of a cargo package version
+#[derive(Debug, PartialEq, Eq)]
+struct CrateVersion {
+    major: u32,
+    minor: u32,
+    patch: u32,
+}
+
+impl PartialOrd for CrateVersion {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for CrateVersion {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        if self.major != other.major {
+            self.major.cmp(&other.major)
+        } else if self.minor != other.minor {
+            self.minor.cmp(&other.minor)
+        } else {
+            self.patch.cmp(&other.patch)
+        }
+    }
+}
+
+/// Parses a crate version string returned from clap
+/// ex: fedimintd 0.3.0-alpha -> 0.3.0
+fn parse_crate_version(version: &str) -> Result<CrateVersion> {
+    // from semver.org's recommended regex
+    // https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    let semver_pattern =
+        Regex::new(r"(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)")?;
+
+    let captured = semver_pattern
+        .captures(version)
+        .ok_or(anyhow!("Not a valid version string"))?;
+
+    Ok(CrateVersion {
+        major: captured["major"].parse()?,
+        minor: captured["minor"].parse()?,
+        patch: captured["patch"].parse()?,
+    })
+}
+
+/// Returns true if the provided version string is greater than or equal to the
+/// min version provided. If there's an error parsing the version, returns
+/// false.
+pub fn is_min_version(
+    min_major: u32,
+    min_minor: u32,
+    min_patch: u32,
+    version_res: Result<String, anyhow::Error>,
```

Comment:
Taking `Result<String, anyhow::Error>` here is odd and confusing.

If for some reason older versions didn't have `--version` or something like that, then the version parsing function should default to some magic number (0.2.0? 0.1.0?) to unify all the cases.

Reply:
Without these issues, we can use some more natural version comparison (I'm not sure if `cmp::Ordering` and operators like `<=` could be used for semver, need more coffee).

BTW. https://docs.rs/semver/latest/semver/ exists and [is not a heavy crate](https://crates.io/crates/semver/1.0.21/dependencies), maybe we should just use it?

Reply:
Good call 👍 

If we're comfortable introducing the semver crate, it simplifies this PR.

https://github.com/fedimint/fedimint/pull/4104/commits/2fa38846037d28a13348254f95eb4d0a12139257